ui/app/controllers/jobs/job/allocations.js

```diff
@@ -40,11 +61,41 @@ export default class AllocationsController extends Controller.extend(
     return ['shortId', 'name', 'taskGroupName'];
   }
 
-  @computed('model.allocations.[]')
+  @computed(
+    'model.allocations.[]',
+    'selectionStatus',
+    'selectionClient',
+    'selectionTaskGroup',
+    'selectionJobVersion'
+  )
   get allocations() {
-    return this.get('model.allocations') || [];
+    const allocations = this.get('model.allocations') || [];
+    const { selectionStatus, selectionClient, selectionTaskGroup, selectionJobVersion } = this;
+
+    if (!allocations.length) return allocations;
```

Comment:
we can just work with `model.allocation` directly here probably - i.e. directly go to the filter.